drivers/gnss/gnss_u_blox_m8.c

```diff
@@ -1,917 +1,671 @@
+UBX_FRAME_DEFINE(disable_gll,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_GLL, 0));
+UBX_FRAME_DEFINE(disable_gns,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_GNS, 0));
+UBX_FRAME_DEFINE(disable_grs,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_GRS, 0));
+UBX_FRAME_DEFINE(disable_gsa,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_GSA, 0));
+UBX_FRAME_DEFINE(disable_gst,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_GST, 0));
+UBX_FRAME_DEFINE(disable_vlw,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_VLW, 0));
+UBX_FRAME_DEFINE(disable_vtg,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_VTG, 0));
+UBX_FRAME_DEFINE(disable_zda,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NMEA_STD, UBX_MSG_ID_NMEA_STD_ZDA, 0));
+UBX_FRAME_DEFINE(enable_nav,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NAV, UBX_MSG_ID_NAV_PVT, 1));
+UBX_FRAME_DEFINE(enable_sat,
+	UBX_FRAME_CFG_MSG_RATE_INITIALIZER(UBX_CLASS_ID_NAV, UBX_MSG_ID_NAV_SAT, 1));
+
+UBX_FRAME_ARRAY_DEFINE(u_blox_m8_init_seq,
+	&disable_gga, &disable_rmc, &disable_gsv, &disable_dtm, &disable_gbs,
+	&disable_gll, &disable_gns, &disable_grs, &disable_gsa, &disable_gst,
+	&disable_vlw, &disable_vtg, &disable_zda, &enable_nav,
 #if CONFIG_GNSS_SATELLITES
-	MODEM_CHAT_MATCH_WILDCARD("$??GSV,", ",*", gnss_nmea0183_match_gsv_callback),
+	&enable_sat,
 #endif
 );
 
-static int ubx_m8_resume(const struct device *dev)
+static void on_pvt_data(struct modem_ubx *ubx, const struct ubx_frame *frame, size_t len,
+			void *user_data)
 {
-	struct ubx_m8_data *data = dev->data;
-	int ret;
-
-	ret = modem_pipe_open(data->uart_pipe, K_SECONDS(10));
-	if (ret < 0) {
-		return ret;
-	}
+	if (len >= UBX_FRM_SZ(sizeof(struct ubx_nav_pvt))) {
+		struct ubx_m8_data *data = CONTAINER_OF(ubx, struct ubx_m8_data, ubx.inst);
+		const struct device *dev = data->dev;
+		const struct ubx_nav_pvt *nav_pvt =
+			(const struct ubx_nav_pvt *)frame->payload_and_checksum;
+
+		if (nav_pvt->time.valid.date &&
+		    nav_pvt->time.valid.time &&
+		    nav_pvt->time.valid.time_fully_resolved) {
+
+			enum gnss_fix_quality fix_quality = GNSS_FIX_QUALITY_INVALID;
+			enum gnss_fix_status fix_status = GNSS_FIX_STATUS_NO_FIX;
+
+			if (nav_pvt->flags.gnss_fix_ok &&
+			    !nav_pvt->nav.flags3.invalid_lon_height_hmsl) {
+
+				switch (nav_pvt->fix_type) {
+				case UBX_NAV_FIX_TYPE_DR:
+				case UBX_NAV_FIX_TYPE_GNSS_DR_COMBINED:
+					fix_quality = GNSS_FIX_QUALITY_ESTIMATED;
+					fix_status = GNSS_FIX_STATUS_ESTIMATED_FIX;
+					break;
+				case UBX_NAV_FIX_TYPE_2D:
+				case UBX_NAV_FIX_TYPE_3D:
+					fix_quality = GNSS_FIX_QUALITY_GNSS_SPS;
+					fix_status = GNSS_FIX_STATUS_GNSS_FIX;
+					break;
+				default:
+					break;
+				}
+			}
```

Comment:
Can parsing like this be moved to a common module, like is done with `gnss_nmea0183_match.c`? I would imagine converting from UBX frames to the types used in the zephyr GNSS API would be the same for any UBX based GNSS modem :) Then you can also fully test the common module once, and each GNSS driver only needs to deal with device specific stuff :)

Reply:
I've pushed an update with a new private module: `gnss_ubx_match.c/.h` with this encapsulated. Hopefully this meets your request!